releases/1.4/kubeflow-bundle.yaml

```diff
@@ -0,0 +1,92 @@
+bundle: kubernetes
+name: kubeflow
+applications:
+  admission-webhook: { charm: admission-webhook, channel: 1.4/stable, scale: 1 }
+  argo-controller: { charm: argo-controller, channel: 3.1.8/stable, scale: 1 }
+  argo-server: { charm: argo-server, channel: 3.0.1/stable, scale: 1 }
+  dex-auth: { charm: dex-auth, channel: 2.28/stable, scale: 1, trust: true }
+  envoy: { charm: envoy, channel: 1.12.2/stable, scale: 1 }
+  istio-ingressgateway:
+    { charm: istio-gateway, channel: 1.5/stable, scale: 1, trust: true }
+  istio-pilot:
+    {
+      charm: istio-pilot,
+      channel: 1.5/stable,
+      scale: 1,
+      options: { default-gateway: 'kubeflow-gateway' },
+    }
+  jupyter-controller:
+    { charm: jupyter-controller, channel: 1.4/stable, scale: 1 }
+  jupyter-ui: { charm: jupyter-ui, channel: 1.4/stable, scale: 1 }
+  katib-controller: { charm: katib-controller, channel: 0.12/stable, scale: 1 }
+  katib-db:
+    {
+      charm: charmed-osm-mariadb-k8s,
+      channel: latest/stable,
+      scale: 1,
+      options: { database: katib },
+    }
+  katib-db-manager: { charm: katib-db-manager, channel: 0.12/stable, scale: 1 }
+  katib-ui: { charm: katib-ui, channel: 0.12/stable, scale: 1 }
+  kfp-api: { charm: kfp-api, channel: 1.7/stable, scale: 1 }
+  kfp-db:
+    {
+      charm: charmed-osm-mariadb-k8s,
+      channel: latest/stable,
+      scale: 1,
+      options: { database: mlpipeline },
+    }
+  kfp-persistence: { charm: kfp-persistence, channel: 1.7/stable, scale: 1 }
+  kfp-profile-controller:
+    { charm: kfp-profile-controller, channel: 1.7/stable, scale: 1 }
+  kfp-schedwf: { charm: kfp-schedwf, channel: 1.7/stable, scale: 1 }
+  kfp-ui: { charm: kfp-ui, channel: 1.7/stable, scale: 1 }
+  kfp-viewer: { charm: kfp-viewer, channel: 1.7/stable, scale: 1 }
+  kfp-viz: { charm: kfp-viz, channel: 1.7/stable, scale: 1 }
+  kubeflow-dashboard:
+    { charm: kubeflow-dashboard, channel: 1.5/stable, scale: 1 }
+  kubeflow-profiles: { charm: kubeflow-profiles, channel: 1.5/stable, scale: 1 }
+  kubeflow-roles:
+    { charm: kubeflow-roles, channel: 1.4/stable, scale: 1, trust: true }
+  kubeflow-volumes: { charm: kubeflow-volumes, channel: 1.4/stable, scale: 1 }
+  metacontroller-operator:
+    {
+      charm: metacontroller-operator,
+      channel: 0.3/stable,
+      scale: 1,
+      trust: true,
+    }
+  mlmd: { charm: mlmd, channel: 1.0/stable, scale: 1 }
+  minio: { charm: minio, channel: 2021-09-03/stable, scale: 1 }
```

Comment:
Minio versioning is terrible, I'm not sure if I like this format for a channel, any kind of fix would make this a lie as it would be released with a new date. I'm tending towards using versioning for our releases e.g. `kf-1.4`, this would give us more flexible. I am open to better ideas though.